types/react/index.d.ts

```diff
@@ -905,51 +906,46 @@ declare namespace React {
     //
     // Event System
     // ----------------------------------------------------------------------
-
-    interface SyntheticEvent<T = Element> {
+    interface BaseSyntheticEvent<E = any, T = any, C = any> {
```

Comment:
Are there better default types for `nativeEvent`, `target` and `currentTarget`? I considered using `Event` and `EventTarget` (from `global.d.ts`), but I wasn't sure if they were appropriate since those are DOM focused.